apps/core/scenes/search/SearchContext.js

```diff
@@ -178,6 +181,18 @@ export default class SearchContextProvider extends React.Component<
     });
   };
 
+  setStateFromQueryParams = (params: Object) => {
+    const parsedParams = qs.parse(params);
+    if (Object.keys(parsedParams).length) {
```

Comment:
Why this check? Is it not sufficient to test `parsedParms != null`?

Reply:
Because `parsedParams` is never null, but `{}`

Reply:
After you give a type to parsedParams, you should not need the if. Just do `Object.values(qs.parse(parsedParams?.travelFrom))`

Reply:
Hmmm `parsedParams` is never null, we need this to only set the state if it actually has something...

Reply:
But as far as I can see, none of these calls will crash if paredParams = {}

Reply:
Well that assigns `travelFrom` and `travelTo` to null instead of the default state. We can conditionally render them though but still need the Object keys check...

Reply:
Well, if `dateFrom` is set but not `travelFrom`, travelFrom will still be null. The check with Object.keys.length does not unsure that travelFrom will be set correctly